server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -379,6 +381,7 @@ void validate(final String key, final Settings settings, final boolean validateD
 
     /**
      * Validates that the settings is valid.
+     * Removing all unknown settings
```

Comment:
This doesn't actually remove them - maybe say 'will ignore unknown settings if the index was created by an older version'

Reply:
Good catch, that's a leftover from my first draft